tools/toolchain/Dockerfile.clang

```diff
@@ -0,0 +1,5 @@
+FROM scylladb/scylla-toolchain:fedora-37-20230111
```

Comment:
This will need updating every time the toolchain is regenerated.

I propose to do it the other way around: make the frozen toolchain depend on the compiler.

```mermaid
graph TD
F37 --> clang-lto
F37 --> toolchain
clang-lto --> toolchain
```
```

Reply:
Although, without the toolchain in clang-lto, how will it build anything?!

Reply:
Maybe we can leap-frog.

1. At time T0, we generate a frozen toolchain without clang+lto (already in the past)
2. At time T1, we generate a frozen toolchain based on T0, +clang+lto
3. At time T2, we need new dependencies, so we generate a frozen toolchain but use `FROM T1 AS ...` to copy the old toolchain.

It seems too complicated.

Reply:
Not sure what to do. On the one hand, I want the build to be self-contained and not depend on older artifacts (which might stop existing). On the other hand, I don't want a multi-hour wait each time I regenerate an image.

Reply:
I don't look at it as multi-hour, but as an overnight process :-/ 

I also don't think we need to regenerate the clang compiler as frequently as we may wish to update the toolchain.

Reply:
We might make the first step use an old toolchain.

```mermaid
flowchart TD
F37 --> dnf1[dnf] --> old[(old-toolchain)]
old --> build-compiler[build compiler] --> compiler[(toolchain+pgo+lto)]
compiler --> copy[copy compiler] --> new[(new toolchain)]
F37 --> dnf2[dnf] --> new
```

Reply:
So, the script to build the compiler will have `FROM some-frozen-toolchain`. The script to build the toolchain will have `FROM some-built-compiler AS ...`. They can be run each in their own cadence.

Reply:
In slightly different domain, but that's what Go team is doing: https://github.com/golang/go/issues/54265, issue is long and refers to others but the story is that:
- they used to build new compiler using some set version (i.e. go 1.4), it worked for 4 years
- they wanted to update this base version to go 1.17
- they've realized it's very hard and took more than a year (!)
- they came to a conclusion that instead of making the base version follow some slow cadence make it a yearly task to keep the process working

disclaimer: I am not volunteering to perform this yearly update task  :smile:

Reply:
The will happen every time we change the compiler, which is at least every 6 months (we had a gap where this was less frequent due to a clang bug which prevented us from upgrading, but we're now back on the treadmill).